### DISCUSSION TITLE
Zim Notebook icons in window lists

### DISCUSSION THREAD

**Discussion by tymmek** [zim-desktop-wiki/zim-desktop-wiki#1996] [Q&A] [answered]
Hello,
I have a question regarding the custom notebook **icons** (those set under _File → Properties_).
I use many separate Zim notebooks, often opening multiple at a time.  In certain environments under Linux (e.g., XFCE, MATE) the custom icon for that notebook appears within the window list, so it's easier to see which notebook I would like to select (the feature seems to work under MS Windows as well):
![xfce](https://user-images.githubusercontent.com/4921863/162061924-53d084fd-a843-4f87-bb5c-e9d8ab155ef1.png)

In others (e.g., Cinnamon/Mint, Gnome, KDE?) every open notebook shares a generic Zim program icon:
![mint-cinnamon](https://user-images.githubusercontent.com/4921863/162061944-f432e7c7-7379-43e4-8d16-a16f23094c26.png)

What particular feature of the desktop environment / window manager allows for displaying these custom icons?  (i.e., How would one go about implementing / requesting such a feature?)

**Comment by introt** [comment, marked as answer]
Hi!

The Zim feature uses [`Gtk.Window.set_icon_from_file`](https://lazka.github.io/pgi-docs/Gtk-3.0/classes/Window.html#Gtk.Window.set_icon), for which the documentation states the following (emphasis mine):

> Sets up the icon representing a [Gtk.Window](https://lazka.github.io/pgi-docs/Gtk-3.0/classes/Window.html#Gtk.Window). This icon is used when the window is minimized (also known as iconified). Some window managers or desktop environments may also place it in the window frame, or display it in other contexts. **On others, the icon is not used at all**, so your mileage may vary.

Since the feature isn't new (here's a SO post from [2013](https://stackoverflow.com/questions/19883234/gtk-window-set-icon-from-file-not-working-properly/19886184#19886184)), not including it seems to be a conscious choice by the DE maintainers. 

Personally, I was not aware of the feature's existence - I've used [MenuLibre](https://github.com/bluesabre/menulibre) to create separate `.desktop` files for my Firefox profiles before stumbling across Zim, so I've just kept doing that for my notebooks as well - besides different icons, this approach also allows you to eg. use [Firejail](https://github.com/netblue30/firejail) for configuration management and the Whisker menu for notebook selection. It does come with a drawback though: notebooks opened via interwiki links get the wrong icon, but I haven't come across the problem too frequently since I just keep them open most of the time.
